pkg/porter/runs.go

```diff
@@ -0,0 +1,118 @@
+package porter
+
+import (
+	"sort"
+	"time"
+
+	"get.porter.sh/porter/pkg/context"
+	"get.porter.sh/porter/pkg/printer"
+	dtprinter "github.com/carolynvs/datetime-printer"
+)
+
+// RunsOptions represent options for showing a particular installation
+type RunsOptions struct {
+	sharedOptions
+	printer.PrintOptions
+}
+
+// Validate prepares for a show bundle action and validates the args/options.
+func (so *RunsOptions) Validate(args []string, cxt *context.Context) error {
+	// Ensure only one argument exists (installation name) if args length non-zero
+	err := so.sharedOptions.validateInstallationName(args)
+	if err != nil {
+		return err
+	}
+
+	err = so.sharedOptions.defaultBundleFiles(cxt)
+	if err != nil {
+		return err
+	}
+
+	return so.PrintOptions.Validate(ShowDefaultFormat, ShowAllowedFormats)
+}
+
+type DisplayRuns []DisplayRun
+
+func (l DisplayRuns) Len() int {
+	return len(l)
+}
+
+func (l DisplayRuns) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+
+func (l DisplayRuns) Less(i, j int) bool {
+	return l[i].Started.Before(l[j].Started)
+}
+
+func (p *Porter) ListInstallationRuns(opts RunsOptions) (DisplayRuns, error) {
+	err := p.applyDefaultOptions(&opts.sharedOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	var displayRuns DisplayRuns
+
+	runs, err := p.Claims.ListRuns(opts.Namespace, opts.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, run := range runs {
+		results, err := p.Claims.ListResults(run.ID)
+		if err != nil {
+			continue
+		}
+
+		displayRun := NewDisplayRun(run)
+
+		if len(results) > 0 {
+			displayRun.Status = results[len(results)-1].Status
+
+			switch len(results) {
+			case 2:
```

Comment:
It is possible to have more than 2 results. You can handle that here by making this case the `default:` case. Then instead of accessing `results[1]` to get the stopped timestamp, you can access the timestamp of the last entry in the array.